src/default-config-backend.cpp

```diff
@@ -9,29 +9,56 @@
 #include <sys/inotify.h>
 #include <unistd.h>
 
-#define INOT_BUF_SIZE (1024 * sizeof(inotify_event))
-static char buf[INOT_BUF_SIZE];
-
+#define INOT_BUF_SIZE (sizeof(inotify_event) + NAME_MAX + 1)
 
 static std::string config_dir, config_file;
 wf::config::config_manager_t *cfg_manager;
 
-static void reload_config(int fd)
+static void readd_watch(int fd)
 {
-    wf::config::load_configuration_options_from_file(*cfg_manager, config_file);
     inotify_add_watch(fd, config_dir.c_str(), IN_CREATE);
     inotify_add_watch(fd, config_file.c_str(), IN_MODIFY);
 }
 
+static void reload_config(int fd)
+{
+    wf::config::load_configuration_options_from_file(*cfg_manager, config_file);
+    readd_watch(fd);
+}
+
 static int handle_config_updated(int fd, uint32_t mask, void *data)
```

Comment:
We might want to also go ahead and check that `mask` has WL_EVENT_READABLE flag set.

Reply:
Is the preferred way to do this `(mask & WL_EVENT_READABLE) != 0`?

Reply:
I'd probably return early if it's not set.

Reply:
Added, though untested.

Reply:
I added a check for `(mask & WL_EVENT_READABLE) == 0` for checking that it's not set. I'll leave this open until further review, because I did the check incorrectly for a split second there.

Reply:
This part looks ok now.